src/io/flutter/editor/WidgetIndentsHighlightingPassFactory.java

```diff
@@ -107,128 +77,37 @@ public void caretPositionChanged(@NotNull CaretEvent event) {
-        if (listener != null) {
-          flutterDartAnalysisService.removeOutlineListener(path, listener);
-        }
+    editorOutlineService.addListener(new OpenEditorOutlineService.Listener() {
+      @Override
+      public void onEditorsChanged() {
       }
 
-      // Register new outline listeners.
-      for (final String path : newPaths) {
-        if (outlineListeners.containsKey(path)) continue;
-        final FlutterOutlineListener listener =
-          (filePath, outline, instrumentedCode) -> {
-            synchronized (outlineListeners) {
-              if (!outlineListeners.containsKey(path)) {
-                // The outline listener subscription was already cancelled.
-                return;
-              }
-            }
-            synchronized (currentOutlines) {
-              currentOutlines.put(path, outline);
+      @Override
+      public void onOutlineChanged(String path) {
+        ApplicationManager.getApplication().invokeLater(() -> {
+          // Find visible editors for the path. If the file is not actually
+          // being displayed on screen, there is no need to go through the
+          // work of updating the outline.
+          if (project.isDisposed()) {
+            return;
+          }
+          for (EditorEx editor : editorOutlineService.getActiveDartEditors()) {
+            if (!editor.isDisposed() && Objects.equals(editor.getVirtualFile().getCanonicalPath(), path)) {
+              runWidgetIndentsPass(editor, editorOutlineService.get(path));
             }
-            ApplicationManager.getApplication().invokeLater(() -> {
-              // Find visible editors for the path. If the file is not actually
-              // being displayed on screen, there is no need to go through the
-              // work of updating the outline.
-              if (project.isDisposed()) {
-                return;
-              }
-              for (EditorEx editor : getActiveDartEditors()) {
-                if (!editor.isDisposed() && Objects.equals(editor.getVirtualFile().getCanonicalPath(), path)) {
-                  runWidgetIndentsPass(editor, outline);
-                }
-              }
-            });
-          };
-        outlineListeners.put(path, listener);
-        flutterDartAnalysisService.addOutlineListener(FileUtil.toSystemDependentName(path), listener);
-      }
-    }
-
-    synchronized (currentOutlines) {
-      for (final String path : obsoletePaths) {
-        // Clear the current outline as it may become out of date before the
-        // file is visible again.
-        currentOutlines.remove(path);
+          }
+        });
       }
-    }
+    });
   }
 
   private void syncSettings(FlutterSettings settings) {
     if (isShowBuildMethodGuides != settings.isShowBuildMethodGuides()) {
       isShowBuildMethodGuides = settings.isShowBuildMethodGuides();
-      updateActiveEditors();
+      // QUESTION: Do we need this?
```

Comment:
I would leave this.  This is a case where the active editors haven't changed but we need to force refresh
 our display as our settings have changed internally.